### PR TITLE
feat(canopy-runoff): wire up output from canopy as input to runoff

### PR DESCRIPTION
**Pull request by langevin-usgs**
* implement minor helper method set_input_to_adapter()
* demonstrate how to wire up output from one box as input to another box

### CODE REVIEW

File: autotest/test_canopy_runoff.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            if "soil_moist" in str(nc_pth):
+                nc_pth = output_dir / "soil_moist_prev.nc"
+            input_variables[key] = nc_pth
+        input_variables["net_ppt"] = None
+        input_variables["net_rain"] = None
+        input_variables["net_snow"] = None
+        runoff = PRMSRunoff(control=control, params=params, **input_variables)
+
+        # wire up output from canopy as input to runoff
+        runoff.set_input_to_adapter("net_ppt", adapter_factory(canopy.net_ppt, "net_ppt"))
```
Comment (jmccreight) [inline autotest/test_canopy_runoff.py:68]: I believe the second variable name can be dropped. AdapterOnedarray only passes that name to super().__init__, which is Adapter and it dosent use it.

Comment (jmccreight) [reply autotest/test_canopy_runoff.py:68]: oops, it does use it to set self._variable. but I'm not sure how useful that is.

Comment (jmccreight) [reply autotest/test_canopy_runoff.py:68]: self._variable is only used by AdapterNetcdf so I think variable_name should only be required for that subclass

File: pynhm/base/adapter.py [hunk 2]
```diff
@@ -40,6 +40,9 @@ def adapter_factory(
     ):
         """Two-D np.ndarrays that are Soltabs"""
```
Comment (jmccreight) [inline pynhm/base/adapter.py:41]: Note to self: I'll remove this soltab subclass from adapter when snow is merged

File: autotest/test_canopy_runoff.py (hunk 1 above)
Comment (langevin-usgs) [reply autotest/test_canopy_runoff.py:68]: will leave as is for now and rethink the wiring as we connect more boxes.

**Pull request merged**